src/main/java/org/csekuaa/backend/controller/UserAccessControlController.java

```diff
@@ -0,0 +1,113 @@
+package org.csekuaa.backend.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.csekuaa.backend.dto.rbac.PermissionDTO;
+import org.csekuaa.backend.dto.rbac.RoleDTO;
+import org.csekuaa.backend.service.UserAccessControlService;
+import org.csekuaa.backend.service.message.ApplicationMessageResolver;
+import org.csekuaa.backend.util.annotation.ADMIN;
+import org.csekuaa.backend.util.annotation.SecureAPI;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/rbac/")
+@RequiredArgsConstructor
+@Tag(name = "User Access Control")
+@SecureAPI
+public class UserAccessControlController {
+    private final UserAccessControlService accessControlService;
+    private final ApplicationMessageResolver messageResolver;
+
+    //add role to user [admin/system admin permission]
+    @PostMapping("user-role/{user_id}/{roleName}")
+    @Operation(summary = "add role to user", description = "only admin role permission user is capable to assign a role to specific user")
+    @ADMIN
+    public ResponseEntity<?> addRoleToUser(@PathVariable("user_id") Integer userId,@PathVariable("roleName") String role) {
```

Comment:
use @Valid for back end validation. As example for Integer : @Valid @Min(value = 0, message = "Value must be positive")